homeassistant/components/bmw_connected_drive/binary_sensor.py

```diff
@@ -28,38 +28,69 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+ALLOWED_CONDITION_BASED_SERVICE_KEYS = [
+    "BRAKE_FLUID",
+    "ENGINE_OIL",
+    "OIL",
+    "TIRE_WEAR_FRONT",
+    "TIRE_WEAR_REAR",
+    "VEHICLE_CHECK",
+    "VEHICLE_TUV",
+]
```

Comment:
Sets are faster if you don't rely on the order of data.

```suggestion
ALLOWED_CONDITION_BASED_SERVICE_KEYS = {
    "BRAKE_FLUID",
    "ENGINE_OIL",
    "OIL",
    "TIRE_WEAR_FRONT",
    "TIRE_WEAR_REAR",
    "VEHICLE_CHECK",
    "VEHICLE_TUV",
}
```